### PR TITLE
*WIP Fix/phpdoc

### PR DESCRIPTION
**Pull request by Dillon-Brown**
## Description

Updated existing PHPDoc to follow standards and fix missing tags.

## How To Test This

See Travis.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [X] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [X] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

**Comment by Mac-Rae** [comment]
Assessed by Travis :+1:

File: Api/V8/OAuth2/Repository/AccessTokenRepository.php [hunk 1]
```diff
@@ -84,7 +84,8 @@ public function persistNewAccessToken(AccessTokenEntityInterface $accessTokenEnt
     /**
      * @inheritdoc
      *
-     * @throws \InvalidArgumentException When access token is not found.
+     * @param $tokenId
+     * @throws \Exception
```
Comment (connorshea) [inline Api/V8/OAuth2/Repository/AccessTokenRepository.php]: 🤔 this seems to remove useful information

File: XTemplate/xtpl.php [hunk 2]
```diff
@@ -374,37 +405,48 @@ function out ($bname) {
...
 
 function SetNullString($str,$varname="") {
 	$this->NULL_STRING[$varname]=$str;
 }
 
-/***[ SetNullBlock ]********************************************************/
-/*
-	sets the string to replace in case the block was not parsed
-*/
+    /***[ SetNullBlock ]*******************************************************
```
Comment (connorshea) [inline XTemplate/xtpl.php]: Isn't this vendored code that we shouldn't mess with?

**Comment by Mac-Rae** [comment]
Assessed by Travis :+1:

File: Api/V8/OAuth2/Repository/AccessTokenRepository.php (hunk 1 above)
Comment (Dillon-Brown) [reply Api/V8/OAuth2/Repository/AccessTokenRepository.php]: Yeah, auto-replace removed it when it replaced the invalid PHPDoc. I can potentially look at adding these back in but at least having valid doc blocks is a step in the right direction imo.

File: XTemplate/xtpl.php (hunk 2 above)
Comment (Dillon-Brown) [reply XTemplate/xtpl.php]: Agree, I must have forgotten to remove the manually included libs before running the PHPDoc fix

File: include/Smarty/plugins/function.sugar_link.php [hunk 3]
```diff
@@ -73,6 +73,9 @@
  * SugarCRM" logo and "Supercharged by SuiteCRM" logo. If the display of the logos is not
  * reasonably feasible for technical reasons, the Appropriate Legal Notices must
  * display the words "Powered by SugarCRM" and "Supercharged by SuiteCRM".
+ * @param $params
+ * @param $smarty
+ * @return string|void
```
Comment (Dillon-Brown) [inline include/Smarty/plugins/function.sugar_link.php]: Lol, yeah I'll need to take another look at this.

**Comment by connorshea** [comment]
@Dillon-Brown the Zend Search Lucene PR being merged has caused many conflicts here. sorry 🙇

**Pull request closed without merging**